src/components/Search/Search.js

```diff
@@ -1,47 +1,146 @@
-import React from 'react';
+import React, { createRef } from 'react';
+import { Link } from 'react-router-dom';
 import './Search.scss';
+import SearchList from '../../components/SearchList/SearchList';
 
-const RECOMMEND_KEYWORD = [
-  '곰표떡볶이',
-  'LA갈비',
-  '육즙만두',
-  '감자탕',
-  '닭갈비',
-];
-const POPULAR_KEYWORD = ['만두', '찹쌀떡', '치즈볼', '도시락', '닭발'];
 class Search extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      productData: [],
+      searchedData: [],
+      inputValue: '',
+    };
+  }
+
+  componentDidMount() {
+    fetch('/data/testData.json')
+      .then(res => res.json())
+      .then(res =>
+        this.setState({
+          productData: res,
+        })
+      );
+  }
+
+  searchByInput = () => {
+    const { productData, inputValue } = this.state;
+    const filtered = productData.filter(data => data.name.includes(inputValue));
+
+    !inputValue
+      ? this.setState({ searchedData: [] })
+      : this.setState({ searchedData: filtered });
+  };
+
+  searchByKeyword = keyword => {
+    const { productData } = this.state;
+    const filtered = productData.filter(data => data.name.includes(keyword));
+
+    this.setState({ searchedData: filtered });
+  };
+
+  handleClickCross = e => {
```

Comment:
- event 를 사용하지 않기 때문에 불필요하겠네요!
```suggestion
  handleClickCross = () => {
```